jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/jobs/QualysDataImporter.java

```diff
@@ -265,6 +266,9 @@ protected List<Map<String, Object>> getHostData(String uriPost, String inputXml)
                                 .collect(Collectors.toList());
                     }
                     break;
+                }else if (resp != null && "UNAUTHORIZED".equals(resp.get("responseCode"))) {
+                    //notify user code
+                    throw new UnAuthorisedException("You are not authorized to access the application through the API");
```

Comment:
UnAuthorisedException is not a run time exception. notification flow will not be triggered.

Reply:
It will be triggered since run time exception is thrown in jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/jobs/HostAssetDataImporter.java

Pls check the JIRA ticket, I have attached notification email as well

Reply:
fetchHostAssets() -> fetchBasedOnInstanceID() -> getHostData()
if exception occurs, presently it is absorbed and the flow continues to fetch assetId from other methods. As per your change, if you throw UnauthorizedException, flow will return to execute() method and program execution stops.

Reply:
the flow is triggered from init